### ISSUE TITLE
Reduce android application size

### ISSUE THREAD

**Issue by LGuilhermeRodrigues** [homuler/MediaPipeUnityPlugin#65]
Hello! I'm using this awesome plugin to build a game and I was thinking about sending it to Play Store.

I've notice the apk size is less than 150MB and that's great. However, when we install the plugin on a smartphone, the app uses 500MB of storage. Is it normal?

I've tried to remove the Mediapipe models that I don't use in the game, keeping only pose tracking, by editing the C/mediapipe_api/BUILD file before make install (in [this repository](https://github.com/LGuilhermeRodrigues/Mediapipe-Invaders/commit/e99a60f2156dc98dbc538247b01fe7b0face3ac0#diff-095319245de28bbebe806edbf125b405882d8a3ab9579fdc755dc4928df58216)) but it think it has changed only 100MB,  from 600MB to 500MB.

Would it be possible to reduce the used space even more? (so kids won't need to uninstall Minecraft to install [my game](https://youtu.be/ILuZ71gyBrI))

PS: Am I using the plugin correctly? I forked the repo and started development on top of it. Or should I copy just some parts and paste on a brand new unity project?

**Comment by homuler** [comment]
Thank you for trying this plugin!
I'll fix it later, but explain it here for convenience.

There are two files that consumes the space: `mediapipe_android.aar` and `mediapipe/models` (AssetBundle).

As to `mediapipe/models`, you can reduce the size by including only models you need.
With only `pose_detection.bytes` and `pose_landmark_upper_body.bytes`, the size would be about 8.7MB (33MB with all models).
If the size of `Assets/StreamingAssets/AssetBundles/mediapipe/models` is much larger, you may forget to delete unused model files under `Assets/MediaPipe/SDK/Models` (by running `make uninstall`).

`mediapipe_android.aar` contains `libmediapipe_jni.so` and `libopencv_java3.so`, and as you did already, you can reduce the size of `libmediapipe_jni.so` by removing unused calculators from its deps.
However, if it's unzipped, `libmediapipe_jni.so` still consumes 259MB (and `libopencv_java3.so` consumes 18MB).

Most of the space is used for symbol information, and you can reduce the size of `libmediapipe_jni.so` by stripping them.
```diff
diff --git a/Makefile b/Makefile
index 31433c4..b733ccf 100644
--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ ifeq ($(OS),Windows_NT)
   bazelflags.cpu += --action_env PYTHON_BIN_PATH="$(PYTHON_BIN_PATH)"
 endif
 bazelflags.android_arm := --config=android_arm
-bazelflags.android_arm64 := --config=android_arm64
+bazelflags.android_arm64 := --config=android_arm64 --linkopt="-s"
 bazelflags.ios_arm64 := --config=ios_arm64 --copt=-fembed-bitcode --apple_bitcode=embedded
 
 proto_srcdir := $(scriptdir)/Protobuf
```

I built the sample app with only Pose Tracking, and the app uses 106MB of the storage (426MB without `--linkopt="s").

**Comment by homuler** [comment]
> PS: Am I using the plugin correctly? I forked the repo and started development on top of it. Or should I copy just some parts and paste on a brand new unity project?

In fact, only files under `SDK` are essential.
Currently the plugin is not provided as a Unity package, and I know that it's not easy to import this plugin to existing projects.
I'll cope this problem later, too.

**Comment by LGuilhermeRodrigues** [comment]
Thanks so much for your response. Now the pose estimation game uses less than 120MB using --linkopt="-s".

**Comment by homuler** [comment]
Fixed by https://github.com/homuler/MediaPipeUnityPlugin/commit/5a01505130725c793ed03df2476b795eb1b6af6c 

Now we can specify `--linkopt=-s` like this.

```sh
python build.py build --android arm64 --strip_all
```

**Comment by GreyPoupwn** [comment]
Is there any matrix / documentation which describes what calculators are required for what tracking models? I'm looking at mediapipe_api/BUILD and not really sure what to comment out!

**Comment by pjbaron** [comment]
The --strip_all parameter seems to have been removed in the current master branch ("unrecognized arguments: --strip_all" using `python build.py build --ios arm64 --strip_all -v`) or maybe it was never available for ios.

It looks like the same thing is available with: `python build.py  build --ios arm64 --linkopt="-s" -v`

**Comment by ROBYER1** [comment]
> Fixed by [5a01505](https://github.com/homuler/MediaPipeUnityPlugin/commit/5a01505130725c793ed03df2476b795eb1b6af6c)
> 
> Now we can specify `--linkopt=-s` like this.
> 
> ```shell
> python build.py build --android arm64 --strip_all
> ```

Could we update the readme docs with this for android at least? It took me a while to figure out how to strip until I properly tried this

**Comment by homuler** [comment]
@pjbaron Thank you!
https://github.com/homuler/MediaPipeUnityPlugin/commit/b8ea5c1fb0b5f0eeebb5de731147d2db46a76e23 fixed https://github.com/homuler/MediaPipeUnityPlugin/issues/65#issuecomment-835934392.
